File: survey_one/Sakif/survey_one.pas
//fpc 3.0.0
//Sakif

program main;

var
    result: Longint;
    i: integer;
function catalan(c: Longint): Longint;    
    function binomial (n,k : Longint): Longint;

        function factorial(num : Longint): Int64;
        begin
            if num <= 0 then
                factorial := 1
            else    
                factorial := (num*factorial(num-1));
        end;
    begin
        binomial:= ((factorial(n) div factorial(k)) div factorial(n-k));
    end;
begin
    catalan := binomial((2*c),c) div (c+1);
end;
begin
    for i:=0 to 9 do
    begin
        result := catalan(i);
        writeln('result=',result);
    end;
end.

File: survey_one/Sakif/survey_one.py
#Sakif
def catalan(n):
    def binomial(n,k):
        def factorial(n):
            if n==0:
                return 1
 
            return n*factorial(n-1)
   
        return factorial(n)/(factorial(k)*factorial(n-k))
        
    return binomial(2*n,n)*(1/(n+1))
def main():
    for n in range(0,10):
        print(catalan(n))

if __name__== "__main__" :
    main()
